token_examples/factory_token/tests/factory_token.rs

```diff
@@ -0,0 +1,191 @@
+            "token constructor returned {:#?}",
+            ret_val
+        );
+    }
+
+    // construct test actors
+    {
+        for actor in [alice, bob] {
+            let ret_val =
+                tester.call_method(operator[0].1, actor, method_hash!("Constructor"), None);
+            assert!(
+                ret_val.msg_receipt.exit_code.is_success(),
+                "actor constructor returned {:#?}",
+                ret_val
+            );
+        }
+    }
+
+    // mint some tokens to alice, who accepts
+    {
+        let ret_val = tester.mint_tokens(
+            operator[0].1,
+            token_actor,
+            alice,
+            TokenAmount::from_atto(100),
+            action(TestAction::Accept),
+        );
+        assert!(ret_val.msg_receipt.exit_code.is_success(), "minting returned {:#?}", ret_val);
+
+        // check balance of test actor, should be zero
+        tester.assert_token_balance(operator[0].1, token_actor, alice, TokenAmount::from_atto(100));
+    }
+
+    // transfer those tokens from alice to bob, who accepts
+    {
+        let params =
+            action_params(token_actor, TestAction::Transfer(bob, action(TestAction::Accept)));
+        let ret_val =
+            tester.call_method_ok(operator[0].1, alice, method_hash!("Action"), Some(params));
+        // check the receipt we got in return data
+        let receipt: Receipt = ret_val.msg_receipt.return_data.deserialize().unwrap();
+        assert!(receipt.exit_code.is_success());
+        // check balances
+        tester.assert_token_balance_zero(operator[0].1, token_actor, alice);
+        tester.assert_token_balance(operator[0].1, token_actor, bob, TokenAmount::from_atto(100));
+    }
+
+    // mint some more to alice, who burns them upon receipt
+    {
+        let ret_val = tester.mint_tokens(
+            operator[0].1,
+            token_actor,
+            alice,
+            TokenAmount::from_atto(100),
+            action(TestAction::Burn),
+        );
+        assert!(
+            ret_val.msg_receipt.exit_code.is_success(),
+            "second minting returned {:#?}",
+            ret_val
+        );
+
+        // check balance of test actor, should be zero
+        tester.assert_token_balance_zero(operator[0].1, token_actor, alice);
+    }
+
+    // disable minting and attempt to mint afterwards
+    {
+        let ret_val =
+            tester.call_method(operator[0].1, token_actor, method_hash!("DisableMint"), None);
+        assert!(
+            ret_val.msg_receipt.exit_code.is_success(),
+            "actor constructor returned {:#?}",
+            ret_val
+        );
+
+        // try minting some tokens, which should fail
+        let ret_val = tester.mint_tokens(
+            operator[0].1,
+            token_actor,
+            alice,
+            TokenAmount::from_atto(100),
+            action(TestAction::Accept),
+        );
+        assert!(
+            !ret_val.msg_receipt.exit_code.is_success(),
```

Comment:
Please assert the actual exit code expected

Reply:
Done, now that it does return a meaningful exit code for each error